### PR TITLE
Updated usaco-991.mdx with Python Code

### PR DESCRIPTION
**Pull request by ravitejaBZ**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-991.mdx [hunk 1]
```diff
@@ -68,4 +68,44 @@ public class Loan {
   
 </JavaSection>
   
+<PySection>
+```
+# our checking function in the binary search
+def isOk(numGallons, withinDays, atLeast, checkNumber):
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:74]: 1. use snake_case
2. type annotations
3. `isOk` tells me nothing about what the function actually does

File: solutions/usaco-991.mdx [hunk 2]
```diff
@@ -70,36 +70,37 @@ public class Loan {
...
-  	    maxMatch = numGallons - (checkNumber * y)
-  	    numDays = (maxMatch - g) // y + 1
-  	    
-  	    if (numDays > withinDays): # take a maximum
-  	        numDays = withinDays
-  	    g += y * numDays # update values
-  	    withinDays -= numDays
-  
-  	return g >= numGallons
+class Loan:
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:73]: why put this in a class?

Comment (ravitejaBZ) [reply solutions/usaco-991.mdx:73]: added class to make the code similar to Java Code which is present.

Comment (SansPapyrus683) [reply solutions/usaco-991.mdx:73]: python != java, so i'd take it out

File: solutions/usaco-991.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-991.mdx]: point 3 isn't addressed

Comment (SansPapyrus683) [reply solutions/usaco-991.mdx]: don't assume the reader knows what "checking" means
what are we checking? what are the parameters? what does it return?
something like "checks whether FJ can [do something] given [stuff]"

Comment (ravitejaBZ) [reply solutions/usaco-991.mdx]: Yeah I've updated and raised a new merge request. please review it once

Comment (ravitejaBZ) [reply solutions/usaco-991.mdx]: @SansPapyrus683 Could you please review my updated changes

Comment (ravitejaBZ) [reply solutions/usaco-991.mdx]: I've added a function description, in the new merge please check out

https://github.com/cpinitiative/usaco-guide/pull/3021/commits/2699376b07180ec4cdfbaeb14062abb4353fab00

Comment (SansPapyrus683) [reply solutions/usaco-991.mdx]: well it isn't present in the PR?

Comment (ravitejaBZ) [reply solutions/usaco-991.mdx]: Sorry I didn't get you

Comment (ravitejaBZ) [reply solutions/usaco-991.mdx]: but if you are asking for the PR ID   the ID is 3021 
  
  https://github.com/cpinitiative/usaco-guide/pull/3021 
  
  the link that I've shared in my previous comment redirects to view the new changes made by me which is in the above PR Mentioned

File: solutions/usaco-991.mdx [hunk 3]
```diff
@@ -68,4 +68,42 @@ public class Loan {
   
 </JavaSection>
   
+<PySection>
+```
+# check whether Farmer John gives Bessie at least N(numGallons) gallons of milk after K(withinDays) days when X == checkNumber
+def isOk(numGallons: int, withinDays: int, atLeast: int, checkNumber: int) -> bool:
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:74]: rename `isOk` to something better
also use snake_case for naming things

File: solutions/usaco-991.mdx [hunk 4]
```diff
@@ -68,4 +68,42 @@ public class Loan {
...
+        g += y * numDays # update values
+        withinDays -= numDays
+    return g >= numGallons
+  
+  
+numGallons, withinDays, atLeast = map(int, input().split())
+low = 1
+high = 1000000000000
+
+while (low<high):
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:97]: space out ops + remove unnecessary parens

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-991.mdx [hunk 5]
```diff
@@ -71,33 +71,28 @@ public class Loan {
 <PySection>
 ```
 # check whether Farmer John gives Bessie at least N(numGallons) gallons of milk after K(withinDays) days when X == checkNumber
-def isOk(numGallons: int, withinDays: int, atLeast: int, checkNumber: int) -> bool: 
+def repay_loan(numGallons: int, withinDays: int, atLeast: int, checkNumber: int) -> bool: 
     g = 0
-    while (withinDays > 0 and g < numGallons):
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx]: use snake_case for all the identifiers

File: solutions/usaco-991.mdx [hunk 6]
```diff
@@ -68,4 +68,37 @@ public class Loan {
...
+            numDays = withinDays
+        g += y * numDays # update values
+        withinDays -= numDays
+    return g >= numGallons
+   
+numGallons, withinDays, atLeast = map(int, input().split())
+low = 1
+high = 1000000000000
+while low<high:
+  	mid = (low+high+1) // 2
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-991.mdx:93]:
```suggestion
while low < high:
  	mid = (low + high + 1) // 2
```

File: solutions/usaco-991.mdx [hunk 7]
```diff
@@ -68,4 +68,37 @@ public class Loan {
...
+        numDays = (maxMatch - g) // y + 1
+        if numDays > withinDays: # take a maximum
+            numDays = withinDays
+        g += y * numDays # update values
+        withinDays -= numDays
+    return g >= numGallons
+   
+numGallons, withinDays, atLeast = map(int, input().split())
+low = 1
+high = 1000000000000
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-991.mdx:91]:
```suggestion
high = 10**12
```

File: solutions/usaco-991.mdx [hunk 8]
```diff
@@ -68,4 +68,37 @@ public class Loan {
...
+while low<high:
+  	mid = (low+high+1) // 2
+  	# we can exclude everything up to but not including mid
+  	if repay_loan(numGallons, withinDays, atLeast, mid):
+  	    low = mid
+  	else: # we can exclude everything including mid
+  	    high = mid -1
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-991.mdx:98]:
```suggestion
  	    high = mid - 1
```

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]

File: solutions/usaco-991.mdx [hunk 9]
```diff
@@ -68,4 +68,37 @@ public class Loan {
   
 </JavaSection>
   
+<PySection>
+```
+# check whether Farmer John gives Bessie at least N(num_gallons) gallons of milk after K(within_days) days when X == check_number
+def repay_loan(num_gallons: int, within_days: int, atLeast: int, check_number: int) -> bool:
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx]: `atLeast` wasn't renamed

**Review by SansPapyrus683 — APPROVED** [review]
@envyaims

**Pull request merged**